tests/chainer_tests/test_variable.py

```diff
@@ -1606,4 +1608,92 @@ def test_loss_scale_gpu(self):
         self.check_loss_scale(cuda.to_gpu(self.x), cuda.to_gpu(self.y))
 
 
+def skip_if_no_ideep():
+    decorator = pytest.mark.skipif(
+        not intel64.is_ideep_available(),
+        reason='ideep is required')
+    return decorator
+
+
+@testing.parameterize(*testing.product({
+    # TODO(niboshi): shape () is not supported
+    'shape': [(0,), (3, 2)],
+    'dtype': [
+        np.int8, np.int16, np.int32, np.int64, np.uint8, np.uint16, np.uint32,
+        np.uint64, np.float16, np.float32, np.float64],
+}))
+@attr.ideep
+class TestIntel64(unittest.TestCase):
+    def setUp(self):
+        self.x_data = np.random.uniform(-1, 1, self.shape).astype(self.dtype)
+
+    def _check_variable_shape_and_dtype(self, var):
+        assert var.data.shape == self.shape
+        assert var.data.dtype == self.dtype
+        assert var.shape == self.shape
+        assert var.dtype == self.dtype
+
+    @skip_if_no_ideep()
```

Comment:
Is this annotation needed? (considering the discussion in https://github.com/cupy/cupy/pull/909)

Reply:
Fixed. Thanks!